group-generators/helpers/data-providers/snapshot/provider.ts

```diff
@@ -216,4 +226,461 @@ export default class SnapshotProvider
+   *
+   * @returns {Promise<FetchedData>} - A Promise that resolves to an object containing fetched data with author addresses as keys and their corresponding vote count as values.
+   */
+  public async querySpaceVotersAboveX({
+    space,
+    abovex,
+  }: QuerySpaceVotersAboveXInput): Promise<FetchedData> {
+    const chunkSize = 1000;
+    let created_gt = 0;
+    let downloadNumber = 0;
+    let currentChunkVoters: { voter: string; created: number }[] = [];
+    const fetchedData: { [address: string]: number } = {};
+
+    do {
+      currentChunkVoters = (
+        await this._querySpaceVotersAboveX(space, created_gt, chunkSize)
+      ).votes;
+
+      for (const currentChunkVoter of currentChunkVoters) {
+        if (!fetchedData[currentChunkVoter.voter]) {
+          fetchedData[currentChunkVoter.voter] = 1;
+        } else {
+          fetchedData[currentChunkVoter.voter] += 1;
+        }
+        created_gt = currentChunkVoter.created;
+      }
+
+      readline.cursorTo(process.stdout, 0);
+      process.stdout.write(`downloading ... (${downloadNumber})`);
+      downloadNumber += currentChunkVoters.length;
+    } while (currentChunkVoters.length > 0);
+
+    const aboveXFetchedData: FetchedData = {};
+
+    if (abovex) {
+      for (const [key, value] of Object.entries(fetchedData)) {
+        if (value >= abovex) {
+          aboveXFetchedData[key] = value;
+        }
+      }
+      return aboveXFetchedData;
+    } else {
+      return fetchedData;
+    }
+  }
+
+  private async _querySpaceVotersAboveX(
+    space: string,
+    created_gt = 0,
+    chunkSize = 1000
+  ): Promise<QueryVotersOutput> {
+    return this.query<QueryVotersOutput>(
+      gql`
+        query GetAllSpaceVoters(
+          $space: String!
+          $created_gt: Int!
+          $chunkSize: Int!
+        ) {
+          votes(
+            first: $chunkSize
+            where: { space: $space, created_gt: $created_gt }
+            orderBy: "created"
+            orderDirection: asc
+          ) {
+            voter
+            created
+          }
+        }
+      `,
+      {
+        chunkSize,
+        space,
+        created_gt,
+      }
+    );
+  }
+
+  public async querySpaceVotersAboveXCount({
+    space,
+    abovex,
+  }: QuerySpaceVotersAboveXInput): Promise<number> {
+    const voters = await this.querySpaceVotersAboveX({ space, abovex });
+    return Object.keys(voters).length;
+  }
+
+  /**
+   * Retrieves proposal authors with votes above a specified number from a specified space and proposal state.
+   * This takes a long time because it needs to query all proposals in SnapShot
+   *
+   * @param {QueryProposalAuthorsAboveXInput} input - Contains parameters for the function:
+   *    @param {string} space - Optional - A string representing the space parameter of the query.
+   *    @param {number} abovex - Optional - A number that defines the minimum number of votes for an author to be included in the result.
```

Comment:
like this no? => A number that defines the minimum number of **proposals created by** an author to be included in the result.

Reply:
Same here - this was explicit in the GitHub issue so I included it

Yes - this was explicit in the GitHub issue so I included it
https://github.com/sismo-core/sismo-hub/issues/1431 

```
{
"name": "number of proposals",
"argName": "TODO",
"type": "TODO",
"example": "3",
"description": "A minimum number of proposals"
},
```

Reply:
Yes so it's : "the minimum number of proposals created by an author" and no "the minimum number of votes for an author" like what you wrote

Reply:
Same here -  I was confused what you were referring to. I'll update the comment